File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/carts/domain/usecase/CartCatalogUseCase.java
package com.tarbadev.witchcraft.carts.domain.usecase;

import com.tarbadev.witchcraft.carts.domain.entity.Cart;
import com.tarbadev.witchcraft.carts.domain.repository.CartRepository;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class CartCatalogUseCase {
    private final CartRepository cartRepository;

    public CartCatalogUseCase(CartRepository cartRepository) {
        this.cartRepository = cartRepository;
    }

    public List<Cart> execute() {
        return cartRepository.findAll();
    }
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/carts/domain/usecase/CartCatalogUseCase.kt
package com.tarbadev.witchcraft.carts.domain.usecase

import com.tarbadev.witchcraft.carts.domain.entity.Cart
import com.tarbadev.witchcraft.carts.domain.repository.CartRepository
import org.springframework.stereotype.Component

@Component
class CartCatalogUseCase(private val cartRepository: CartRepository) {

    fun execute(): List<Cart> {
        return cartRepository.findAll()
    }
}
